### DISCUSSION TITLE
Exactly what features will I be locked out of if I refuse to use 2FA?

### DISCUSSION THREAD

**Discussion by kjkrum** [community/community#140222] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

The email says I'll be required to use 2FA because I "contribute code" to Github. Which features will I be unable to use if I refuse? I already deleted most of my repos and moved my projects to another site after the notification feed debacle. I rarely submit PRs. The only thing I really use this site for is writing bug reports on other people's repos.

**Comment by feijoes** [comment]
As stated in the [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/about-mandatory-two-factor-authentication), you will not be able to access GitHub.com until you enable 2FA. If you attempt to access GitHub.com, you will be prompted to enable 2FA

**Comment by iamasadshah** [comment]
Hey there! I can understand why the 2FA requirement might feel like a hassle, especially if you’re not contributing code frequently. I’ll break down what happens if you don’t enable 2FA, and I’ll provide some documentation references from GitHub to help clarify things.

### Features You’ll Lose Without Enabling 2FA

If GitHub has flagged your account for 2FA because you contribute code, it’s part of their effort to improve security for contributors. Here’s what will happen if you choose not to enable 2FA:

1. **Loss of Access to Code Contributions**:  
   According to GitHub's documentation, if you’re required to enable 2FA but choose not to, you’ll lose the ability to push code to repositories or create new repositories within any organization that requires 2FA. Even if you only contribute occasionally (e.g., bug reports, pull requests), these activities could be blocked.
   
   - [GitHub 2FA Requirement Documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/about-two-factor-authentication#2fa-requirements)

2. **Access to Writing Bug Reports**:  
   If you're mostly submitting **bug reports**, those might still be allowed since creating issues or commenting on repositories doesn’t necessarily require code contributions. However, if you're part of an organization or contributing to repos where 2FA is required, you might lose access to even these features depending on the repo’s policies.

3. **GitHub Organizations**:  
   If you’re part of any GitHub organization that enforces 2FA, you won’t be able to remain a member of that organization without enabling 2FA. You’ll be removed from those orgs, and any privileges like issue tracking or discussions will be restricted.

   - You can find more details here: [GitHub Organization 2FA Policy](https://docs.github.com/en/organizations/managing-organization-settings/enforcing-two-factor-authentication-for-your-organization)

### Why GitHub is Pushing for 2FA
GitHub has become more strict about security, especially after some incidents. Even if you’re not actively pushing code, enabling 2FA helps secure the entire platform, especially for open-source projects where many contributors are involved.

### What to Do Next
If your usage is limited to writing bug reports and you don’t want to enable 2FA, I suggest checking the individual repositories to see if they have 2FA as a requirement for all contributors. Otherwise, you might still be able to use GitHub for reporting bugs and issues without needing 2FA, but pushing code or being part of organizations will definitely require it.

I hope this clears up the situation! Let me know if you have any more questions, and I can help you navigate GitHub’s requirements further.

**Comment by kjkrum** [comment]
Sounds like I may need to create a new account and never publish code so it does not nag me to enable 2FA. Third party repos that require 2FA to write bug reports will simply not get the benefit of receiving bug reports. This will hasten the death of Github as the default place to share code.

**Comment by kjkrum** [comment]
#67053
